### PR TITLE
Add roles for Docker and Consul

### PR DESCRIPTION
**Pull request by kberzinch**
- Install Docker following the [Docker on CentOS install guide](https://docs.docker.com/engine/install/centos/)
- Install Consul
- Bootstrap Consul ACLs following the [HashiCorp guide](https://developer.hashicorp.com/consul/tutorials/day-0/access-control-setup)

### CODE REVIEW

File: roles/consul/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    path: /var/opt/nomad/run/
+    state: directory
+    mode: '0777'
+
+- name: Start Consul service
+  ansible.builtin.systemd_service:
+    name: consul
+    state: started
+    enabled: false
+  when: (consul_hcl['content'] | b64decode | regex_search('[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}')) == None
```
[the remark this answers is not in the document]
Comment (kberzinch) [reply roles/consul/tasks/main.yml:53]: Yes, the Consul token format is UUIDs (unless you do something HashiCorp discourages)

File: README.md [hunk 2]
```diff
@@ -1,2 +1,27 @@
...
+      ansible_user: gburdell3
+      ansible_become: true
+      datacenter: bcdc
+      node_name: bcdc1
+```
+
+Run the playbook like so:
+
+```sh
+ansible-playbook --inventory inventory.yml playbook.yml
```
[the remark this answers is not in the document]
Comment (kberzinch) [reply README.md:26]: It looks like it only checks `/etc/hosts` by default, but I could create a config file to do something else. I'll fiddle around with it later.

File: roles/consul/tasks/main.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (kberzinch) [reply roles/consul/tasks/main.yml:53]: I would like to factor out that parsing to a variable but I couldn't figure out how to do it cleanly

File: roles/docker/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    state: started
+    enabled: true
+
+- name: Wait for Docker to be available
+  ansible.builtin.wait_for:
+    path: /var/run/docker.sock
+    state: present
+    timeout: 10
+
+- name: Run Hello World image
```
Comment (evan10s) [inline roles/docker/tasks/main.yml:52]: is this just to test?

**Review by evan10s — APPROVED** [review]

File: roles/docker/tasks/main.yml (hunk 3 above)
Comment (kberzinch) [reply roles/docker/tasks/main.yml:52]: Yes

File: README.md [hunk 4]
```diff
@@ -23,5 +23,5 @@ platform:
 Run the playbook like so:
 
 ```sh
-ansible-playbook --inventory inventory.yml playbook.yml
+ansible-playbook playbook.yml
```
Comment (evan10s) [inline README.md:26]: what goes in playbook.yml?

Comment (kberzinch) [reply README.md:26]: That's a file that exists in this branch

Comment (evan10s) [reply README.md:26]: why is it also in gitignore then? or am i missing something

Comment (kberzinch) [reply README.md:26]: oh whoops, will fix

Comment (kberzinch) [reply README.md:26]: I meant to gitignore the inventory not the playbook, pushed fix

Comment (evan10s) [reply README.md:26]: ok i see, now my question is what goes in inventory.yml, does it make sense to include an example?

Comment (kberzinch) [reply README.md:26]: There's an example in the README itself; mostly it's the basic inventory fields (host, user, become) plus two (for now) variables `datacenter` and `node_name` for plugging into Consul and eventually Nomad configuration.

Idk if I'll commit the actual file. I'm still mulling over what things are secrets and how those might be managed. Either way it won't be until we actually have a RHEL9 box managed with this playbook.

Comment (evan10s) [reply README.md:26]: seems fine to me

**Review by RossIV — APPROVED** [review]

**Pull request merged**